base/pending_task.cc

```diff
@@ -22,7 +22,7 @@ PendingTask::PendingTask(const Location& posted_from,
       delayed_run_time(delayed_run_time),
       leeway(leeway),
       delay_policy(delay_policy) {
-  if (!recordreplay::AreEventsDisallowed()) {
+  if (!recordreplay::AreEventsDisallowed() && !recordreplay::AreEventsPassedThrough()) {
```

Comment:
Q: In what type of situation should we check for both?

Reply:
Ideally the AreEventsDisallowed() call would be enough, but when code can run on non-recorded threads we also have to check AreEventsPassedThrough().  It would be nice to improve this.